builder/builder.go

```diff
@@ -443,3 +453,88 @@ func parseImageNameFromArgs(cmdArgs string) string {
 	}
 	return cmdArgs[:idx]
 }
+
+// prepareVolumeSource creates and populates the host file and volume for the specified source type
+func (b *Builder) prepareVolumeSource(ctx context.Context, volMount *volume.Volume) error {
+	switch {
+	case volMount.Source.Secret != nil:
+		if err := b.createSecretFiles(ctx, volMount); err != nil {
+			return err
+		}
+		return b.populateSecretVolume(ctx, volMount)
+	default:
+		return errors.New("volume source type not supported yet")
+	}
+}
+
+// createSecretFiles creates necessary files for source type Secret
+func (b *Builder) createSecretFiles(ctx context.Context, volMount *volume.Volume) error {
+	var args []string
+	var sb strings.Builder
+	if runtime.GOOS == util.WindowsOS {
+		args = []string{"powershell.exe", "-Command"}
+		sb.WriteString("mkdir " + volMount.Name)
+		log.Println("making directory " + volMount.Name)
+	} else {
+		args = []string{"/bin/sh", "-c"}
+		sb.WriteString("mkdir " + volMount.Name + " && ")
+	}
+	for k, v := range volMount.Source.Secret {
+		val := v
+		decoded, err := base64.StdEncoding.DecodeString(val)
+		if err != nil {
+			return errors.New("failed to decode Base64 value. please make sure value provided is Base64 encoded")
+		}
+		val = string(decoded)
+		if runtime.GOOS == util.WindowsOS {
+			sb.WriteString("; Add-Content -Path ")
+			sb.WriteString(volMount.Name + "/" + k)
+			sb.WriteString(" -Value @\"\n")
+			sb.WriteString(val)
+			sb.WriteString("\n\"@")
+		} else {
+			sb.WriteString("cat >> ")
+			sb.WriteString(volMount.Name + "/" + k)
+			sb.WriteString(" <<EOL\n")
+			sb.WriteString(val)
+			sb.WriteString("\nEOL\n")
+		}
+	}
+	args = append(args, sb.String())
+	var buf bytes.Buffer
+	if err := b.procManager.Run(ctx, args, nil, &buf, &buf, ""); err != nil {
+		return errors.Wrapf(err, "failed to write value, %s", buf.String())
+	}
+	log.Println("buffer: ", buf.String())
+	log.Println("Created new file(s) for volume: ", volMount.Name)
```

Comment:
@northtyphoon do we want to log this? or this sounds like an implementation detail?